BUILD.gn

```diff
@@ -26,6 +26,10 @@ import("${chip_root}/gn/chip/tests.gni")
 import("${chip_root}/gn/chip/tools.gni")
 
 if (current_toolchain != "${dir_pw_toolchain}/dummy:dummy") {
+  declare_args() {
+    enable_python_build_tools = true
```

Comment:
It's a python module, and a tool, but not a build tool. Maybe chip_enable_python ?

Can we move to tools.gni, and

assert(chip_enable_python)

in src/controller/python/BUILD.gn ?

Reply:
I am still unclear about how our things are layed out. I can disable python globally. For now it seems the only place we use this python bit is under `if (chip_build_tools)` hence the extra checl.

Do you think python global disabling would be more correct? 
I would like to keep the cpp sources to compile under controller/python since those should also be checked by LGTM.com.

Reply:
Also

`true` -> `current_os == "linux"`

Reply:
> I am still unclear about how our things are layed out. I can disable python globally. For now it seems the only place we use this python bit is under `if (chip_build_tools)` hence the extra checl.

It's not possible to disable python altogether as its used for code generation. chip_enable_python could mean "disable compiling the python module or modules" however. Maybe it's too confusing and it should be chip_build_python or chip_enable_python_modules  instead.

My objection is just to the name - enable_python_build_tools parses as "enable the python build tools" with build as a noun.

chip_build_tools is meant as "[chip] build the tools" with build as a verb. This wasn't talking about build tools, but building tools.

> 
> Do you think python global disabling would be more correct?
> I would like to keep the cpp sources to compile under controller/python since those should also be checked by LGTM.com.

Reply:
Renamed to 'chip_enable_python_modules'.